### PR TITLE
Add S3RequestSplit processor groovy script

### LINKED ISSUE #36: Convenience script to split FetchS3Object requests
Large files in S3 can be range requested.  There is an update pending for FetchS3Object in NIFI-7886 with PR at github.com / apache / nifi / pull / 4576 to expose an ability to do this by setting properties in the flow.  To leverage this for large files a convenience script will be prepared.

### PR DESCRIPTION
**Pull request by lucasmoten**
- Includes documentation and tests
- Intended to take output of ListS3 as input to prepare for FetchS3Object with range requests

Closes #36

### CODE REVIEW

File: nifi-script-processors/S3RequestSplit.groovy [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    // ### Remove the initiating flowfile that is now split up -----------------
+    session.remove(ff)
+  }
+} catch (e) {
+  // Log to console
+  log.error("Error splitting s3 requests: ${aS3Bucket}${aFilename}", e)
+  // Set attribute with error
+  ff = session.putAttribute(ff, 's3requestsplit.error.message', e.toString())
+  // Transfer
+  session.transfer(ff, REL_FAILURE)
```
Comment (TheMover) [inline nifi-script-processors/S3RequestSplit.groovy:205]: As redundant as it sounds I think there should be a safeguard on operations inside this catch block such as another try/catch statement, there are a few operations within the catch block that have potential exceptions which won't be caught and could effectively stop the processor.

File: nifi-script-processors/S3RequestSplit.groovy [hunk 2]
```diff
@@ -0,0 +1,206 @@
+import groovy.io.FileType
+import org.apache.commons.io.IOUtils
+import java.nio.charset.StandardCharsets
+
+// ###########################################################################
+// S3 Request Split
+//
+// This is an auxiliary script that can be used as part of a NiFi flow to 
+// break up requests for larger files fromn S3 into smaller segments.
```
Comment (TheMover) [inline nifi-script-processors/S3RequestSplit.groovy:9]: typo on `files fromn S3` should be `files from S3`

**Review by TheMover — APPROVED** [review]
Tests pass, looks good to me, left a few comments that could be turned into issues but I won't hold up the review for them.

File: nifi-script-processors/S3RequestSplit.groovy (hunk 1 above)
Comment (lucasmoten) [reply nifi-script-processors/S3RequestSplit.groovy:205]: Actually thats a good point. For the ExecuteGroovyScript processor that these get laid down in, there is no guarantee that a "failure" relationship will exist.  By default the processor sets up for rollback and retry, so hitting this catch could log, and then fail to send to failure.

Comment (lucasmoten) [reply nifi-script-processors/S3RequestSplit.groovy:205]: That scenario is true of all the Groovy script based processors

**Review by SethLasky — APPROVED** [review]
LGTM

**Review by TheMover — APPROVED** [review]

**Pull request merged**